drivers/timer/riscv_machine_timer.c

```diff
@@ -8,8 +8,15 @@
 #include <spinlock.h>
 #include <soc.h>
 
-#define CYC_PER_TICK ((uint32_t)((uint64_t)sys_clock_hw_cycles_per_sec()	\
-			      / (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC))
+#if CONFIG_RISCV_MACHINE_TIMER_SYSTEM_CLOCK_DIVIDER == 0
+#define CYC_PER_TICK ((uint32_t)((uint64_t)sys_clock_hw_cycles_per_sec() \
+				 / (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC))
+#else
+#define CYC_PER_TICK ((uint32_t)((uint64_t) (sys_clock_hw_cycles_per_sec() \
+				 >> CONFIG_RISCV_MACHINE_TIMER_SYSTEM_CLOCK_DIVIDER) \
```

Comment:
if 0, shift will be a no-op, so why 2 definitions?

Reply:
It is intended to reduce the effect on the typical case, but I was too worried.
There seems to be no impact, so we will integrate it.